jgrapht-core/src/main/java/org/jgrapht/generate/DirectedScaleFreeGraphGenerator.java

```diff
@@ -0,0 +1,406 @@
+ * certain number of edges is reached irrespective to the number of nodes.<br>
+ * However, because the target number of edges is not known beforehand, in this implementation, we
+ * added another feature that enables the user to grow the curve according to that power low until
+ * the target number of edges or target number of nodes is reached.
+ * 
+ * @author Amr ALHOSSARY
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class DirectedScaleFreeGraphGenerator<V, E>
+    implements
+    GraphGenerator<V, E, V>
+{
+    private final Random rng;
+
+    /**
+     * probability that the new edge is from a new vertex v to an existing vertex w, where w is
+     * chosen according to d_in + delta_in criterion.
+     */
+    private final float alpha;
+    /**
+     * The probability that the new edge is (from a new vertex v to an existing vertex w) plus the
+     * probability that the new edge is (from an existing vertex v to an existing vertex w), where v
+     * and w are chosen independently, v according to d_out + delta_out, and w according to d_in +
+     * delta_in criteria. This equals 1 - gamma. Gamma refers to the probability that the new edge
+     * is from an existing vertex v to a new vertex w.
+     */
+    final float alphaPlusBeta;
+
+    /** In-degree bias used for Alpha and Beta */
+    private final float deltaIn;
+    /** Out-degree bias used for Beta and Gamma */
+    private final float deltaOut;
+    /**
+     * Target total number of edges to reach. It has a higher priority than {@link #targetNodes}.
+     * Zero is a valid value.<br>
+     * If negative number, the user does not care about the total number of edges and is interested
+     * only in the number of nodes, therefore, {@link #targetNodes} will be considered instead.
+     * Otherwise, {@link #targetEdges} will be considered and {@link #targetEdges} will be ignored.
+     */
+    private final int targetEdges;
+    /**
+     * Target total number of targetNodes to reach.<br>
+     * This has lower priority than {@link #targetEdges}. It will not be used unless
+     * {@link #targetEdges} given is a <i>negative</i> number.
+     */
+    private final int targetNodes;
+
+    /**
+     * An enum to indicate the vertex selection using its inDegree or outDegree
+     */
+    private static enum Direction
+    {
+        IN,
+        OUT
+    };
+
+    /**
+     * Maximum number of consecutive failed attempts to add an edge.
+     */
+    private int maxFailures = 1000;
+
+    /**
+     * Control whether the generated graph may contain loops.
+     */
+    private boolean allowingMultipleEdges = true;
+
+    /**
+     * Control whether the generated graph many contain multiple (parallel) edges between the same
+     * two vertices
+     */
+    private boolean allowingSelfLoops = true;
+
+    /**
+     * Constructs a Generator with a Random Number Generator using a random seed.
```

Comment:
The seed is not random. It is usually constructed from the system time. 

Anyway, just change to "Construct a new generator".